src/com/sap/piper/analytics/Telemetry.groovy

```diff
@@ -45,53 +46,37 @@ class Telemetry implements Serializable{
     }
 
     protected static void piperOsDefaultReporting(Script steps, Map payload) {
-        try {
-
-            def swaCustom = [:]
-
-            /* SWA custom parameters:
-                custom3 = step name (passed as parameter step)
-                custom4 = job url hashed (calculated)
-                custom5 = build url hashed (calculated)
-                custom10 = stage name
-                custom11 = step related parameter 1 (passed as parameter stepParam1)
-                custom12 = step related parameter 2 (passed as parameter stepParam2)
-                custom13 = step related parameter 3 (passed as parameter stepParam3)
-                custom14 = step related parameter 4 (passed as parameter stepParam4)
-                custom15 = step related parameter 5 (passed as parameter stepParam5)
-            */
-
-            def swaUrl = 'https://webanalytics.cfapps.eu10.hana.ondemand.com/tracker/log'
-            def idsite = '827e8025-1e21-ae84-c3a3-3f62b70b0130'
-            def url = 'https://github.com/SAP/jenkins-library'
+        def swaEndpoiont = 'https://webanalytics.cfapps.eu10.hana.ondemand.com/tracker/log'
+        Map swaPayload = [
+            'idsite': '827e8025-1e21-ae84-c3a3-3f62b70b0130',
+            'url': 'https://github.com/SAP/jenkins-library',
+            'action_name': payload.actionName,
+            'event_type': payload.eventType,
+            'custom3': payload.step,            // custom3 = step name (passed as parameter step)
+            'custom4': payload.jobUrlSha1,      // custom4 = job url hashed (calculated)
+            'custom5': payload.buildUrlSha1,    // custom5 = build url hashed (calculated)
+            'custom10': payload.stageName       // custom10 = stage name
+        ]
+        // step related parameters
+        for(def key : [1, 2, 3, 4, 5]){         // custom11 - custom15 = step related parameter 1 - 5 (passed as parameter stepParam1 - stepParam5)
+            if (payload["stepParam${key}"] != null) swaPayload.put("custom1${key}", payload["stepParam${key}"])
+        }
 
-            swaCustom.custom3 = payload.step
-            swaCustom.custom4 =  payload.jobUrlSha1
-            swaCustom.custom5 = payload.buildUrlSha1
-            swaCustom.custom10 = payload.stageName
-            swaCustom.custom11 = payload.stepParam1
-            swaCustom.custom12 = payload.stepParam2
-            swaCustom.custom13 = payload.stepParam3
-            swaCustom.custom14 = payload.stepParam4
-            swaCustom.custom15 = payload.stepParam5
+        def payloadString = swaPayload
+            .collect { entry -> return "&${entry.key}=${URLEncoder.encode(entry.value.toString(), "UTF-8")}" }
+            .join('')
+            .replaceFirst('&','?')
```

Comment:
[minor] [style] IMO we should avoid doing things wrong and fix afterwards. Here have `&` and afterwards fix the first occurance to be `?`.  I would like to suggest to first have the `?` in the payloadString and afterwards `collect{./.}. without the `&` together with join('&').